### PR TITLE
[mongodb] authenticate with X.509

### PR DESCRIPTION
**Pull request by yannmh**
To authenticate using a X.509 certificate, one must use the `db.auth()`
method specifying `MONGODB-X509` for the `mechanism` field, and the user
that corresponds to the client certificate for the user field.

More information:
https://api.mongodb.com/python/current/examples/authentication.html
https://docs.mongodb.com/manual/tutorial/configure-x509-client-authentication/

### CODE REVIEW

**Comment by degemer** [comment]
Could you also add a `try ... except` around the `authenticate` method ?
According to the [docs](https://api.mongodb.com/python/current/api/pymongo/database.html?highlight=authenticate#pymongo.database.Database.authenticate) it can raise a subclass of `PyMongoError`, and we don't catch it for now (thus not sending the SC).

**Comment by degemer** [comment]
Apart from that, LGTM

File: checks.d/mongo.py [hunk 1]
```diff
@@ -533,6 +534,38 @@ def _normalize(self, metric_name, submit_method, prefix):
...
+                authenticated = database.authenticate(username, mechanism='MONGODB-X509')
+
+            # Username & password
+            else:
+                authenticated = database.authenticate(username, password)
+
+        except pymongo.errors.PyMongoError as e:
+            self.log.error(
+                u"Authentication failed due to invalid credentials or configuration issues. %s", e
+            )
```
Comment (degemer) [inline checks.d/mongo.py]: I think there's a `raise` missing.

Comment (degemer) [inline checks.d/mongo.py]: Could this log.error leak the `password` if there was an auth failure ?

Comment (degemer) [inline checks.d/mongo.py]: For posterity's sake: raise is not needed as it is dealt with once the function returns `False`.
And @yannmh is working in another branch on avoiding credentials leak in all logs.

**Comment by olivielpeau** [comment]
This was released with 5.9.0, removing 5.10 milestone

**Pull request merged**